Fails to build in rawhide due to PCP API change

See https://koji.fedoraproject.org/koji/taskinfo?taskID=25213880
```
gcc -DHAVE_CONFIG_H -I.  -I. -I./src -I. -I./src -DSRCDIR=\"/builddir/build/BUILD/cockpit-162\" -DBUILDDIR=\"/builddir/build/BUILD/cockpit-162\" -DDATADIR=\"/usr/share\" -DPACKAGE_LIBEXEC_DIR=\""/usr/libexec"\" -DPACKAGE_SYSCONF_DIR=\""/etc"\" -DPACKAGE_BIN_DIR=\""/usr/bin"\" -DPACKAGE_LOCALSTATE_DIR=\""/var/lib/cockpit"\" -DPACKAGE_LIB_DIR=\""/usr/lib64"\" -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT   -I./src/bridge -DG_LOG_DOMAIN=\"cockpit-pcp\" -I/usr/include/gio-unix-2.0/ -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread  -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_34 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_34 -I/usr/include/json-glib-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread    -Wall           -Werror=strict-prototypes -Werror=missing-prototypes           -Werror=implicit-function-declaration           -Werror=pointer-arith -Werror=init-self           -Werror=format=2           -Werror=return-type           -Werror=missing-include-dirs -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -mcet -fcf-protection -g -c -o src/bridge/test_pcp_archives-test-pcp-archives.o `test -f 'src/bridge/test-pcp-archives.c' || echo './'`src/bridge/test-pcp-archives.c
In file included from /usr/lib64/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from src/bridge/cockpitchannel.h:23,
                 from src/bridge/cockpitmetrics.h:23,
                 from src/bridge/test-pcp.c:22:
src/bridge/test-pcp.c: In function 'init_mock_pmda':
src/bridge/test-pcp.c:49:13: error: implicit declaration of function '__pmLocalPMDA'; did you mean 'pmSpecLocalPMDA'? [-Werror=implicit-function-declaration]
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
             ^~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:379:8: note: in definition of macro '_G_BOOLEAN_EXPR'
    if (expr)                                    \
        ^~~~
/usr/include/glib-2.0/glib/gtestutils.h:118:49: note: in expansion of macro 'G_LIKELY'
                                              if G_LIKELY (expr) ; else \
                                                 ^~~~~~~~
src/bridge/test-pcp.c:49:3: note: in expansion of macro 'g_assert'
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
   ^~~~~~~~
src/bridge/test-pcp.c:49:28: error: 'PM_LOCAL_CLEAR' undeclared (first use in this function); did you mean 'PM_FATAL_ERR'?
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
                            ^~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:379:8: note: in definition of macro '_G_BOOLEAN_EXPR'
    if (expr)                                    \
        ^~~~
/usr/include/glib-2.0/glib/gtestutils.h:118:49: note: in expansion of macro 'G_LIKELY'
                                              if G_LIKELY (expr) ; else \
                                                 ^~~~~~~~
src/bridge/test-pcp.c:49:3: note: in expansion of macro 'g_assert'
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
   ^~~~~~~~
src/bridge/test-pcp.c:49:28: note: each undeclared identifier is reported only once for each function it appears in
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
                            ^~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:379:8: note: in definition of macro '_G_BOOLEAN_EXPR'
    if (expr)                                    \
        ^~~~
/usr/include/glib-2.0/glib/gtestutils.h:118:49: note: in expansion of macro 'G_LIKELY'
                                              if G_LIKELY (expr) ; else \
                                                 ^~~~~~~~
src/bridge/test-pcp.c:49:3: note: in expansion of macro 'g_assert'
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
   ^~~~~~~~
src/bridge/test-pcp.c:50:28: error: 'PM_LOCAL_ADD' undeclared (first use in this function); did you mean 'PF_LOCAL'?
   g_assert (__pmLocalPMDA (PM_LOCAL_ADD, 333, "./mock-pmda.so", "mock_init") >= 0);
                            ^~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:379:8: note: in definition of macro '_G_BOOLEAN_EXPR'
    if (expr)                                    \
        ^~~~
/usr/include/glib-2.0/glib/gtestutils.h:118:49: note: in expansion of macro 'G_LIKELY'
                                              if G_LIKELY (expr) ; else \
                                                 ^~~~~~~~
src/bridge/test-pcp.c:50:3: note: in expansion of macro 'g_assert'
   g_assert (__pmLocalPMDA (PM_LOCAL_ADD, 333, "./mock-pmda.so", "mock_init") >= 0);
   ^~~~~~~~
cc1: some warnings being treated as errors
make[1]: *** [Makefile:6351: src/bridge/test_pcp-test-pcp.o] Error 1
```

--- a/src/bridge/mock-pmda.c
+++ b/src/bridge/mock-pmda.c
@@ -74,12 +74,19 @@ static int64_t counter64 = INT64_MAX - 100;
 static int
 mock_fetchCallBack(pmdaMetric *mdesc, unsigned int inst, pmAtomValue *atom)
 {
+#if PM_VERSION_CURRENT >= PM_VERSION(4,0,0)
+  if (pmID_cluster(mdesc->m_desc.pmid) != 0)
+    return PM_ERR_PMID;
+
+  switch (pmID_item(mdesc->m_desc.pmid)) {
+#else
   __pmID_int		*idp = (__pmID_int *)&(mdesc->m_desc.pmid);
 
   if (idp->cluster != 0)
     return PM_ERR_PMID;
 
   switch (idp->item) {
+#endif
   case 0:
     if (inst != PM_IN_NULL)
       return PM_ERR_INST;

--- a/src/bridge/test-pcp.c
+++ b/src/bridge/test-pcp.c
@@ -46,8 +46,13 @@ init_mock_pmda (void)
       exit (0);
     }
 
+#if PM_VERSION_CURRENT >= PM_VERSION(4,0,0)
+  g_assert (pmSpecLocalPMDA ("clear") == NULL);
+  g_assert (pmSpecLocalPMDA ("add,333,./mock-pmda.so,mock_init") == NULL);
+#else
   g_assert (__pmLocalPMDA (PM_LOCAL_CLEAR, 0, NULL, NULL) >= 0);
   g_assert (__pmLocalPMDA (PM_LOCAL_ADD, 333, "./mock-pmda.so", "mock_init") >= 0);
+#endif
 
   void *handle = dlopen ("./mock-pmda.so", RTLD_NOW);
   g_assert (handle != NULL);

